app/router/student.py

```diff
@@ -3,52 +3,91 @@
+)
 from id_generation import JNVIDGeneration
 from request import build_request
 from helpers import db_request_token
 
 router = APIRouter(prefix="/student", tags=["Student"])
 
 
+def build_student_exam_data(data):
+    # this function builds the student-exam record object
+    student_exam_data = {}
+    for key in data.keys():
+        if key in mapping.STUDENT_EXAM_RECORD_QUERY_PARAMS:
+
+            # the key 'student_id' is the ID entered by the user but the 'student_id' being stored in the student-exam table is the PK of a row in the student table.
+            # Hence, we retreive the PK of a student based on their 'student_id' and that is stored in the student-exam record table.
+            if key == "student_id":
+                student_response = get_students(
+                    build_request(query_params={"student_id": data["student_id"]})
+                )
+                student_exam_data["student_id"] = int(student_response[0]["id"])
+
+            # for the key 'exam_name', we have to retrieve the respective exam PK to store in the student-exam record table
+            elif key == "exam_name":
+                exam_response = exam_router.get_exam(
+                    build_request(query_params={"name": data[key]})
+                )
+                if len(exam_response) != 0:
+                    student_exam_data["exam_id"] = int(exam_response[0]["id"])
+
+            # for any other key, we store the value as the user entered
+            else:
+                student_exam_data[key] = data[key]
+    return student_exam_data
+
+
 def build_enrollment_data(data):
+    # this function builds the enrollment data object of a student
     enrollment_data = {}
     for key in data.keys():
         if key in mapping.ENROLLMENT_RECORD_PARAMS and key != "student_id":
             enrollment_data[key] = data[key]
     return enrollment_data
 
 
-def build_student_data(data):
+async def build_student_data(data):
+    # this function builds the student data object
     student_data = {}
     for key in data.keys():
         if key in mapping.STUDENT_QUERY_PARAMS:
-            student_data[key] = data[key]
+
+            # the key 'planned_competitive_exams' contains an array of exam names. Each exam PK is retrieved and stored in the student table.
+            if key == "planned_competitive_exams":
+                exam_ids = []
+                for exam in data[key]:
+                    exam_response = exam_router.get_exam(
+                        build_request(query_params={"name": exam})
+                    )
+                    if exam_response != []:
+                        exam_ids.append(exam_response[0]["id"])
+
+                student_data[key] = exam_ids
+
+            # for any other key, we store the value as the user entered
+            else:
+                student_data[key] = data[key]
     return student_data
 
 
 def build_user_data(data):
+    # this function builds the user data object
```

Comment:
why is this function in student route? 

also, in USER_QUERY_PARAMS there is a key `check_consent` -- its not there in db-service